news/7998.bugfix

```diff
@@ -0,0 +1 @@
+Ask for password when either the password is None.
```

Comment:
This probably needs to be reworded. It's not very clear what this means as a line in https://pip.pypa.io/en/stable/news/.

I'm not sure what the best phrasing would be, but perhaps someone else would have a suggestion.

Reply:
How about `Prompt the user for password  if the backend keyring doesn't return one`